### DISCUSSION TITLE
User experience test - results

### DISCUSSION THREAD

**Discussion by matkoniecz** [streetcomplete/StreetComplete#4264] [Feedback]
I [again](https://github.com/streetcomplete/StreetComplete/issues/1470) run an UX test on a small sample of ten people, and reviewed OSM notes, posted discussions and SC-related topics in attempt to find similar feedback.

I tried to fix/report what I found as a separate issues but taking additional actions or creating separate actionable issues may be worth doing based on the info here. Feel free if you notice that something like that is missing.

Note: UX sample was small, biased and not representative of anything except how willing they were to help me. Nevertheless I learned or relearned multiple things. Tests were done almost entirely during Winter and Spring.

------------------------------------------------------

Overall, there is a noticeable improvement compared to what people expressed previous UX testing. Some people were tested again (they participated in #1470 testing round) and they noted, unprompted, noticeable improvements.

Testing was a bit less productive - because number of issues encountered by new mappers was noticeably lower.

One of the received comments was

> I am unsure is there a point in meeting, I installed app and everything is clear

Nevertheless, multiple new issues were spotted and many were fixed.

Other selected comments (also translated):

> Repeating recent used level in building level quest is great

> Classifying and filling forms is too similar to my job and I would never do this as a hobby

> This link collection is great, I discovered some projects thanks to it

Some people expressed doubt and confusion about some quests "why this info is collected?". The only serious negative comments that were appearing were complaints about (1) collecting very unimportant data and (2) heavy battery use.

----------------------------------------------------------------

Reminder, especially to me and @westnordost : regular people are confused among things which are obvious to OSM mappers.

For example, multiple people idependently asked about additional hints that would help to distinguish between various surface values as they were really unsure - and some of them made mistakes.

Some test mappers failed to distinguish flat and gray `surface=dirt` from `surface=concrete`! Not sure how much SC can help here, that is case of skills which not really used in general.

The single garage also needs an explanation (selected where building=garages was more fitting). Maybe show also `=garages` in the initial top values prefill. Or do not show `=garage` there

Regular people are confused by many this that seem obvious to people tagging them for a long time.
"Oh, so these dots before crossings are for blind people" "oh so that is why traffic lights are beeping"
I observed one person entering wrong bicycle parking capacity, reading the hint and correcting it.

So in case of doubt, I would rather add an extra explanation.

One of the mappers in the kerb height selection quest selected one that was matching by material - not by height. (concrete vs stone)
On asking why they did this they noticed the problem.

----------------------------------------------------

Show all quests for an element at once (#124) was the top requested change by people - in various forms. "autoopen next quest if the element has multiple quests" or "show all tasks for the element" were the most common. Sadly, I do not really have an idea how to do this well.

----------------------------------------------------

The road tap interface (sidewalk quest, cycleway quest) is the second most confusing part of SC

https://github.com/streetcomplete/StreetComplete/discussions/33643364 - specifically about the incline quest (this part is fixed!)

People are highly confused by it. Out of 7 people who encountered it - 7 were confused. One managed to forget how to use it within a few days and was confused again when we tested the app again. One person out of 7 managed relatively quickly to discover how to use it - and they are also a programmer and proceeded to debug the form.

How to solve it?

Show crude popup "click here"?
A colourful bubble on the sidewalk itself "click me"?

Show the message below the question explaining how this interface should be used?

Show popup for new users?

Design a completely new interface? Especially https://github.com/streetcomplete/StreetComplete/discussions/2684 seems interesting and worth considering, but that would leave the cycleway and sidewalk surface even more unique and confusing...

Lane selection for double no like with sidewalks or cycleways is extremely confusing. People are regularly confused that "no" should be selected at all and that it should be done on both sides of the road.

Often not even reaching "can't say', some selected no for one side and considered the question as answered - and were confused about what they should do now.

----------------------------------------------------------------

I was pessimistic about the usability of the opening hours editor, but people used it without a real trouble. Also for resurvey quests. 

----------------------------------------------------------------

I was pessimistic about the lit quest during the day but it turned even worse: every single tested person who encountered `lit` quest during day was confused, many people were entering the wrong data when surveying during the day.

I would strongly support moving it to an overlay-only task. Or get back to disabling it during the day, with an option to override it by power users.

Right now is is the most confusing part of StreetComplete.

----------------------------------------------------------------

One of the universally observed things: surveying building levels is really confusing, time-consuming and dominates other tasks.

It is quite depressing to see how much effort is getting consumed by this quest.

I would seriously consider disabling it by default. This quest is confusing, extremely time-consuming, with limited benefits, often unclear or hard to answer and boring.

It is quite sad to see how much effort it consumes, its enablement by default is likely discouraging potential mappers, and overall reducing the value of quest mapped. Disabling it by default would result in an immediate drop, but I expect that overall greater retention would be helpful - especially when quests would be weighted by value).

<s>I know that I proposed disabling it by default already and it got rejected but after testing I am really sure that it would be a good idea.</s>

EDIT: this appears to be wrong, it seems I have never proposed it

EDIT2: see #3295 for examples of problems with classifying building level count

EDIT3: disabled in https://github.com/streetcomplete/StreetComplete/commit/0bcd6d71b04eb57f30ff4c69905d910847d83e68

----------------------------------------------------------------

Also, the orchard quest is really hard to answer. I am above average among city dwellers in plant recognition and I can identify several common plants (including wheat/rye/etc from each other). But I am unable to differentiate between pear trees and apple trees. And cherry trees.

Also, none of the people who run into the orchard quest could answer its question.

I think that it is another candidate for being disabled by default.

EDIT: orchard quest disabled by default in edb8a9dc4 (it can be still enabled if you want to solve it!)

----------------------------------------------------------------

It is not clear that the building=* quest is not asking about current use. Or about the most prominent public use. People keep misinterpreting this question and look through entries and comment "there is no entry here for greengrocer" when they see an apartment building with a greengrocer at the floor level. I do not have a good idea how to improve things further here.

----------------------------------------------------------------

I was pessimistic about the usability of the opening hours editor, but people used it without real trouble. Also for resurvey quests. 

----------------------------------------------------------------

implemented:

* stop asking for roof shape by default #3161 https://github.com/streetcomplete/StreetComplete/pull/3161
* #3341 - discussed with some actions taken (for example replacing the image of unmarked crossing). Maybe more needs to be done, let's see how changes done will affect users. 
* small tweaks: #3635 #3443 #3383 #3328 a8b73d78b2a9c54ab5e67994587c99bbc97d490f #3427 #3402 #3421 #3390 #3576 462e287aac823dc185cade87e9a6baa953906b58 https://github.com/streetcomplete/streetcomplete-mapstyle/pull/130
 including things reported by others and treated with higher priority as a result of UX search such as #4174
* other minor upstream tweaks: https://github.com/openstreetmap/id-tagging-schema/issues/523 ( https://github.com/openstreetmap/id-tagging-schema/pull/524 ), https://github.com/openstreetmap/id-tagging-schema/pull/528
* https://github.com/openstreetmap/id-tagging-schema/issues/529 was done to ensure that SC is not affected by other similar issues (though nothing major was found)
* improving styling of the ranking page (#4203 #3414) was a result of previous UX tests
* Multiple improvements in Polish translation (typos, misleading translations, tortured grammar, inconsistent terminology)
* #4246 (minor achievement tweak) and #4215 #3387 (surprisingly time-consuming building quest interface improvement, still definitely worth it) are implemented but still in the PR phase
* #4105 and https://josm.openstreetmap.de/ticket/22102#comment:9 - better tracktype-surface conflict handling in JOSM (merged!) and StreetComplete (in the PR phase)
* #3429 #3473 - better description for parking quest
* probably something else which I missed
* https://taginfo.openstreetmap.org/projects/streetcomplete#tags - taginfo listing (#4225)
* better images for ramp quest (partially done for copyright-related results - but see also https://github.com/streetcomplete/StreetComplete/blob/7d5ab65024a057f23eab9325c73088239097ac6b/app/src/main/res/drawable-xxhdpi/ramp_bicycle.jpg vs https://github.com/streetcomplete/StreetComplete/blob/55287f240eee2d06cf9a47629c69094b71b90b67/app/src/main/res/drawable-xxhdpi/ramp_bicycle.jpg ), with time on this split between this tasks

reported and implemented by others:

* https://github.com/streetcomplete/streetcomplete-mapstyle/issues/107
* Also, triggered fix in a proprietary (but replaceable) dependency ( via https://github.com/streetcomplete/streetcomplete-mapstyle/issues/108 )

rejected:

* #3394 stop asking crossing detail for unmarked crossings
* #4058 Provide extra info allowing to identify a specific object such as bicycle parking
* #4172 move address quest higher in the list to make more clear that important data is also being asked about

stuck:

* https://github.com/streetcomplete/StreetComplete/discussions/3440 (change "Ground (generic)" to "Natural ground"? But that will encourage using it instead of more specific values)
* https://github.com/streetcomplete/StreetComplete/discussions/3442 - linking used tags. Doing this properly requires listing tags used by StreetComplete (which also enables taginfo project listing). This one turned out to be surprisingly complex! #4225 was implemented partially to enable it - but it revealed some issues in how OSM Wiki handles compound tags and some things need to be resolved there first before such linking will be a good idea. See https://wiki.openstreetmap.org/wiki/Talk:Wiki#Redundant_pages_for_%22building_block%22_tags ("Enhanced not-found messages for non-English users" and "name:mos" sections are relevant blockers)

not counted as part of UX work time for grant purposes, but still a direct result of UX investigation:

* https://github.com/openstreetmap/id-tagging-schema/issues/533 https://github.com/openstreetmap/id-tagging-schema/issues/547 https://github.com/openstreetmap/iD/issues/9209 https://github.com/openstreetmap/id-tagging-schema/issues/531 https://github.com/openstreetmap/iD/issues/9209#issuecomment-1192727593
* triggered multiple edits in Poland to reduce variants of names in operator tags for ATMs (editing time here was not counted toward UX work, so it was not a paid editing)

----------------

My work on UX testing was sponsored by a NGI Zero Discovery grant.

See #1470 for the previous one.

**Comment by westnordost** [comment]
> The single garage also needs an explanation (selected where building=garages was more fitting). Maybe show also =garages in the initial top values prefill. Or do not show =garage there

Wording is already "single garage" and shows a picture of one car. A garage for two cars (on the grounds of a detached house) would still count as a "single garage" too, so a description like "Garage for a single car" would not be correct. I think it is fine / low priority.

> The road tap interface (sidewalk quest, cycleway quest) is the second most confusing part of SC
> How to solve it?
>  A colourful bubble on the sidewalk itself "click me"?

Well, what is done now is to show the "tap here" hint, as known from other apps. (The ripple.) That is hopefully a good hint for many to just try to tap it. When they tapped it, hopefully they would get it because it is WYSIWYG.

Another solution would be to disable all those quests (sidewalk, cycleway, ...), which is something I thought about anyway for all quests where there is also a layer that does the same. (Because using the layer would be more efficient and would be the preferred way to do it.)

> I was pessimistic about the lit quest during the day but it turned even worse: every single tested person who encountered lit quest during day was confused, many people were entering the wrong data when surveying during the day.

See above, it could be disabled because there is a layer.

> One of the universally observed things: surveying building levels is really confusing, time-consuming and dominates other tasks.
> It is quite depressing to see how much effort is getting consumed by this quest.
> I would seriously consider disabling it by default. This quest is confusing, extremely time-consuming, with limited benefits, often unclear or hard to answer and boring.

Ok. Needs description that should mention that it is time-consuming and sometimes difficult to answer (if such a description does not exist yet). But anyway, can you link to the issue/discussion where your prior suggestion was rejected? Maybe I forgot some good arguments to keep it enabled by default.

> Orchard
> I think that it is another candidate for being disabled by default.

But it does not really occur that often, does it? And it occurs for probably relatively large areas, so it's not spamming the view in that sense.

**Reply by matkoniecz** [reply]
> But anyway, can you link to the issue/discussion where your prior suggestion was rejected? Maybe I forgot some good arguments to keep it enabled by default.

Apparently I was misremembering. I should have linked it from start as it appears to be not existing. I found only  #3169 which was about power poles.

> But it does not really occur that often, does it? And it occurs for probably relatively large areas, so it's not spamming the view in that sense.

It happens often enough that I have decided to test SC only in the city to avoid running into this question (I guess I could also disable in on the phones used by testers...). At least in my area there is large number of orchards that are half-abandoned/abandoned but still existing. And it is not getting better for modern industrial orchards, as these are rarely signed with info what they produce and types of trees are not recognizable at all.

https://overpass-turbo.eu/s/1kIT - note that I am quite often in this area, I am trying to answer SC quests and still orchard quest remains not answered there :(

![screen24](https://user-images.githubusercontent.com/899988/182778078-9299fe90-e572-4806-ad64-e480ca52de3a.png)

> Well, what is done now is to show the "tap here" hint, as known from other apps. (The ripple.)

Yes, I know. Sadly people are not really noticing it.

> I think it is fine / low priority.

I agree, it mostly goes on pile "remember that people are more confused that you would expect"

**Reply by matkoniecz** [reply]
> Ok. Needs description that should mention that it is time-consuming and sometimes difficult to answer (if such a description does not exist yet). 

`override val defaultDisabledMessage = R.string.default_disabled_msg_difficult_and_time_consuming`

gives

> This quest type is disabled by default because it is either time-consuming or difficult to make the right choice. Only enable this quest type if you are really committed to adding this kind of data.

I plan on applying it in a few days, unless someone convinces me otherwise.

**Reply by matkoniecz** [reply]
> Another solution would be to disable all those quests (sidewalk, cycleway, ...), which is something I thought about anyway for all quests where there is also a layer that does the same. (Because using the layer would be more efficient and would be the preferred way to do it.)

Actually, sidewalk one works mostly fine and I would consider keeping it. But I would disable lit quest by default, while allowing users to reenable it if they want. I also plan on doing this, though here a new message may be needed to say something specifically about difficulty of solving it during the day.

**Reply by westnordost** [reply]
Ok, then maybe the orchard one can also be disabled by default.

**Reply by mpeter50** [reply]
> > The road tap interface (sidewalk quest, cycleway quest) is the second most confusing part of SC
> > How to solve it?
> > A colourful bubble on the sidewalk itself "click me"?
> 
> Well, what is done now is to show the "tap here" hint, as known from other apps. (The ripple.) That is hopefully a good hint for many to just try to tap it. When they tapped it, hopefully they would get it because it is WYSIWYG.

For me the ripple effect only plays when I release a lane after tapping, but only then. It doesn't play after opening the quest's menu and just waiting.
Is it expected to work this way? If so, I think it's not effective, as the user won't get hinted that they can tap there, but maybe theres a bug here.

**Reply by riQQ** [reply]
For me it (one side slowly getting darker and then brighter again, then the same for the other side) only shows the first time opening such a quest dialog after starting StreetComplete. Is this intentional or a bug?

**Reply by mpeter50** [reply]
That's right, this is what actually happens. The animation is showing for me too, but only once, and it does not repeat if I reopen the quest or if I open a different instance of this quest, but only an app restart.
App restart here means force closing, closing it with pressing the back button until it disappears is not enough.

**Reply by westnordost** [reply]
It's intentional. It only shows the first time.

**Comment by lgommans** [comment]
> Some people expressed doubt and confusion about some quests "why this info is collected?". The only serious negative comments that were appearing were complaints about (1) collecting very unimportant data and (2) heavy battery use.

Virtually all of my friends and family wonder why I bother with doing this a lot of the time. Some things make sense, like adding whether a parking lot is paid: that's something they would like to see on a map. A lot of other things make sense if you think for two seconds: we might not care about kerb heights, but wheelchair users do. Wheelchair is never mentioned in that quest.

And then there is the category where I also have no idea myself, like building stories and roof shape. (I've got those disabled.) Is that for 3d renders? You'd still not know in which orientation the roof is and I usually can't see the backside so I'd just be guessing about that part. What about stairs' step count: who could use that data? Would there be an app that lets you set a cut-off point for how many steps you're willing to walk before it tells you there is no path to the destination? Wouldn't the elevation difference be more important (tall steps are more exhausting) and way easier to obtain (for entire countries at once, rather than individual stairs' data)?

Not everyone cares to hear my guess/assumption that things like road surface go into routing algorithms as tie-breaker (seems unimportant to answer for every single piece of road, iff that's the only purpose), but even with people that care, I can't explain many quests' importance adequately.

Having template answers, either for myself or to answer questions, would be quite nice. Seeing more value might also lead to more contributions.

This could tie in with info on how to use the quest (like the sidewalk, cycleway UI). Such a pop-up page could also give more info about the different options that didn't fit into the main UI, common pitfalls, and/or info shown for expert users to say which button corresponds to which tag value (when I was new to the app, I would sometimes know which tag to use but not which image to click for that; other times, I'm just curious to learn how to tag something outside of SC). A question mark in the corner of each quest, which opens that info page in a local HTML view thing, could lead people to this extra info.

I'd be happy to research and contribute this information for each quest in case this would be something you decide is a good idea.

---

> Some test mappers failed to distinguish flat and gray `surface=dirt` from `surface=concrete`

> I observed one person entering wrong bicycle parking capacity, reading the hint and correcting it.

> One of the mappers in the kerb height selection quest selected one that was matching by material - not by height. (concrete vs stone). On asking why they did this they noticed the problem.

> every single tested person who encountered `lit` quest during day was confused  

Would these people be mappers under other circumstances? This seems a little extreme :sweat_smile:  
And please don't make `lit` day-only again, I'm not often driving on public transport at night since it doesn't exist here (sad), so I wasn't getting anything done whereas I can recognize a street lantern-lined street during the day just fine... If need be, an expert option here would work for everyone, though if I wonder about it being patronizing and if the real target audience would understand this more intuitively.

**Reply by matkoniecz** [reply]
> Would these people be mappers under other circumstances? This seems a little extreme

I still remember how confused I was by some things when I was starting to map!

Though yes, some of this people were not really interested in mapping and wanted to help me or were bribed with ice cream (like my younger sister and her friends). So average SC mapper is likely less confused.

But especially `lit` quest was confusing even to people who had no problem with other things :/

> And then there is the category where I also have no idea myself, like building stories and roof shape. (I've got those disabled.) Is that for 3d renders?

Yes, and floor count may be used to identify building type and as proxy for population density.

> What about stairs' step count: who could use that data?

Navigation for blind people would be able to announce how many steps are expected. And routing can detect and assign lower penalties for steps with very low density (200 m of `highway=steps`, only 20 `step_count`). And wheelchair routing can also allow for example `step_count=1` and `step_count=2`.

> Not everyone cares to hear my guess/assumption that things like road surface go into routing algorithms as tie-breaker

For routing `surface=unhewn_cobblestone` and `surface=sand` is not only tie-breaker :)

**Reply by lgommans** [reply]
> bribed with ice cream

Can you also bribe me please? :smile: 

> For routing `surface=unhewn_cobblestone` and `surface=sand` is not only tie-breaker :)

I was thinking that no road would be reasonably considered for routing anyway if it's cobblestone or sand, they'd not be the path that most people travel and so not even qualify for highway=tertiary and so the router never picks it anyway.

But yeah for something like cycling, it might not be so picky and go for shorter routes rather than fewer turns / higher maxspeeds. If that ends the cyclist up on a sandy road that is impossible to cycle on, they would indeed curse the router :P

I've added a little section here experimentally. Do you think it's a good idea to collect these on the wiki? Whether it eventually ends up inside SC on an info card is less important then. https://wiki.openstreetmap.org/wiki/Key:surface#Uses

I'm not entirely sure about this setup, maybe a central page makes more sense. Perhaps it could have it reversed: per goal (like routing) it might have an itemized list of tags used for it (and on that line, you can also specify how/why if it's not necessarily obvious).

**Comment by rugk** [comment]
## LIt quest

>I was pessimistic about the lit quest during the day but it turned even worse: every single tested person who encountered lit quest during day was confused, many people were entering the wrong data when surveying during the day.
>
>I would strongly support moving it to an overlay-only task. Or get back to disabling it during the day, with an option to override it by power users.
>
>Right now is is the most confusing part of StreetComplete.

What does it mean they entered it wrong?

I personally just look whether there is a lamp and if so, yes. Maybe I look whether it has this [red sign thing](https://de.wikipedia.org/wiki/Laternenring), then I know (because I have learned it through StreetComplete!) what to do, but otherwise "Yes".

**Reply by matkoniecz** [reply]
They answered `no` after not spotting some lamps - in one case place is lit by gigantic lamps providing lighting primarily to nearby expressway, in another lamps are on buildings or wires between buildings.

**Reply by rugk** [reply]
I guess they did this, because they think the lamps need to be nearby/next to the street, instead of checking _that_ the way is actually somewhat _lit_ i.e. and thus passable at night.

Another solution would thus be to change the text to focus of the question asked to sth. like this:
> **Is this way passable at night?**
> _A way is passable at night if it is lit by nearby lamps or other light sources._

(variant 0)

### Variant 1

would also allow the red-sign thing and make users aware there is an "other answer" they may need to chose:
>** Is this way passable at all night?**
> _A way is passable at night if it is lit by nearby lamps or other light sources all the night. Note signs may show when it is not lit the whole night._

(longish text though)

### Variant 2
Also note we may wanna not make it that strong/confusing, as other factors may include whether it is passable

> **Is this way lit and [thus] passable at night?**
> _If a sign indicates it is not lit the whole night, please choose an answer at "other answers"._

Also "passable" may be the rong term here I dunno.

Or so.

**Reply by andrewharvey** [reply]
"Is there enough lighting to see well on a dark night".

Many people including myself interpret the current quest as does this way have it's own dedicated lights, regardless of if they actually do a good job at lighting the way or if there is nearby lights spilling over.

**Reply by lgommans** [reply]
I don't know if the word "passable" is applicable here, as virtually everyone carries a portable flashlight nowadays. Sure you can pass it, not like it floods overnight, but what is really being asked is whether there is light to see by available there.

Andrew's suggestion sounds good to me, but your interpretation of the quest is... I want to say "wrong", but honestly it's a valid interpretation of the quest text, just not what the `lit` tag means. Wiki says:

> indicates the presence of lighting

I think it's broad enough to mean: whether the object is lit up at night, by whatever light (be it belonging to the object itself, such as a bus stop, or a light in the vicinity that shines sufficiently on the object and isn't likely to change in the future).

**Reply by mamoffel** [reply]
The quest and the wiki description are very good in their brevity, but where can one find help for interpretation?

For me it would help if there were an assistent asking even simpler questions along a flow-chart like decision tree and guiding me to the right answer. In case of the lit-quest an assistent could ask for the presence of nearby lamps showing some examples (lamps along the way, nearby lit expressway, lamps hanging on wires between buildings and so on), then ask if it is lit all the night and so on.

I personally couldn't find any advice of this kind even on the web. Hence it is sometimes a very time consuming intellectual task to get the question and the expected result right.

**Reply by matkoniecz** [reply]
@mamoffel 

If you can improve wiki documentation that would be highly welcome!

If you have good examples of photos where it is unclear how to make tagging you can also post it on the talk page of OSM Wiki.

There were also some ideas how to implement such mini tutorial in StreetComplete, I think there is an open issue concerning it.

**Comment by milh0use** [comment]
+1 for disabling the building levels quest by default. It took me a while to find that some quests can be turned off and the building levels bubbles obscure what I think could more usefully be collected by SC. Around me, most roads are long terraces of identical height buildings, recording their height is much more efficiently done with a bulk select in JOSM.
Regarding editing opening times, the easiest I've found is in Every Door, it uses several common defaults that make entry really quick, such as:
- default days M-F (checkboxes), opening time as a tap on 06:00, 06:30, 07:00, 07:30...09:00, 09:30 or custom (pick an hour, then pick minutes past that hour), closing time as another list of half hourly times from 16:00-19:00 or custom, and when you've entered the first rule, if any days were unchecked it prompts you for a second rule with those missing days selected, and you can choose "closed" or go through the whole opening time/closing time process again. Process continues until every day of the week is covered by a rule.

**Reply by westnordost** [reply]
As to your suggestion regarding opening hours:

> default days M-F (checkboxes)

If you mean that M-F are already checked by default: This is already the case. Actually, even better, it checks the **workdays** of the week, e.g. in Iran it is Sa-Th. In different countries, there are different workdays.

> and when you've entered the first rule, if any days were unchecked it prompts you for a second rule with those missing days selected

This is already the case.

> opening time as a tap on 06:00, 06:30, 07:00, 07:30...09:00, 09:30 or custom

Possible. But awkward for businesses that e.g. only open in the evening. Even if you just cover 6:00 to 10:00 as starting times, with a 30 minute interval, you have 10 buttons in that dialog, that may be a bit too much. It's only a few buttons less that the numerical input keyboard usually has. Also, what about shops that close for lunch in EveryDoor?

Before doing any experiments here, maybe it would be more prudent to focus on #3796

**Reply by siiky** [reply]
Commenting on the opening hours input dialogues, I much prefer SC over ED because the correct hours interval is rarely any of the ones ED suggests, so I pretty much always have to hit the "more hours" (or something) button. (I haven't been counting but I would say less than 1/10)

I also find that the hours/minutes are a lot harder to find on ED than on SC exactly because it doesn't use a clock looking thing. Sure it's sorted and all, but I don't want to be binary searching the thing.

In short, the SC clock thingy feels a lot more natural to use the ED, and ED rarely suggests the interval I'm looking for. And it's a lot of ckicking like @westnordost said.

**Reply by mpeter50** [reply]
> > and when you've entered the first rule, if any days were unchecked it prompts you for a second rule with those missing days selected
> 
> This is already the case.

Is it? I can click "add opening hours" once again, and select "Add days of the week", but days which don't yet have opening hours defined are not checked by default.

**Comment by mpawelski** [comment]
I'm a bit sad that building level quest was disabled by default :( 
I rarely had issue with it and I think the UI for it was great. I really liked it because I saw that the answer I provide is actually used by apps like Organic Maps and maps.me to render building heights.
Occasionally I had troubles with answering this question (multilevel building, I was to close to the building and couldn't count the levels, etc) but I think more general functionality to "skip this quest" would be much more valuable. Such "skip" button would also be great for other quests where it's impossible/hard to answer (for example because of visibility) for particular object.

**Reply by FloEdelmann** [reply]
You can already skip quests via "Other answers …" → "Can't say" → "Skip".

And, just for the record, you can still manually re-enable the building levels quest (and others) in the settings.

**Reply by mpawelski** [reply]
ah, I thought I need to write some note if I want to skip it, but I see it's not necessary... ignore my comment then.

**Comment by mpeter50** [comment]
> The road tap interface (sidewalk quest, cycleway quest) is the second most confusing part of SC

> How to solve it?

Maybe the lanes could slowly blink, gradually change the brightness of their background, or something similar, so it would call attention at the detail that the lanes are to be clicked. In case you do this, I think the different lanes should blink separately (e.g. delayed, or one lane blinking at a time, maybe the blinking can be semi-random so that a lane blinks every 5-10 seconds).
Or as it was mentioned, a ripplce effect could work too, and it would also be nicer than what I suggested.

> Often not even reaching "can't say', some selected no for one side and considered the question as answered - and were confused about what they should do now.

I think this might be a problem for all quest types.
Maybe the user could be served tips, for example that this button exists in the "other answers..." menu.
I've also written some more ideas below.

> I was pessimistic about the usability of the opening hours editor, but people used it without a real trouble. Also for resurvey quests.

Some thoughts of mine:

In my country it is unusual to have the same opening hours for saturday as for weekdays, so having saturday ticked here does not make much sense.
I've seen above that it was mentioned that the pre-ticked days are determined by the workdays of the country, but is it determined by the country where the building is located, or by the region setting of the user's phone?

When on the go, I have problems with the time picker dialog.
When picking the hour, clicking one of the values in the inner circle (so values from 13 to 23 and 00) almost always results in a different one getting selected.
Cannot reproduce this right now (it just does not occur now, no idea why because otherwise it actually always happens), but wanted to mention it. Also, if this is the time picker from one of the official support libraries, possibly this cannot be changed either.

Also, this quest has important very specific options in the "other answers..." menu, one of which can be "correctly" specified by selecting the proper timeframe. I can easily imagine that users are picking 0:00-0:00 for every day instead of using 24/7, or since they don't know they can input differing opening hours by month they simply just input the opening hours that applies to the month when they surveyed.
Remember, they even had a similar problem with the road tap interface: they didn't find the can't say option.
It seems for some reason the "other answers..." button is not natural for users to open when in doubt. Maybe they don't see it (in that case it could have a slightly different background color or an icon) or it's not unambigous that there are additional special answers (in that case the button could be renamed)?

> I was pessimistic about the lit quest during the day but it turned even worse: every single tested person who encountered lit quest during day was confused, many people were entering the wrong data when surveying during the day.

~~What were the problems exactly?~~ In the meantime I have read back and.. yeah I have been considering doing that quest only at night for the exact same reason: I can only determine for sure if something is not lit when I see that its dark.

Personally I think it's also sometimes hard to decide on what should I select at night, as some streets have street lighting, but lights are rare enough that between every 2 lights there are several meters on the road that is very dark, so much so that I wouldn't feel safe riding my scooter at normal speeds there because I wouldn't be able to see where is the road damaged, or if theres some other obstacle.
In such cases sometimes I partition the street and mark accordingly, but I don't really know how should I do this, when there are parts of the street that is just not lit _properly_. And this is not everything, because I'm only looking at the "litness" road, sidewalks are usually even darker, especially if they are covered by trees and bushes.

**Reply by matkoniecz** [reply]
> In my country it is unusual to have the same opening hours for saturday as for weekdays, so having saturday ticked here does not make much sense.
> I've seen above that it was mentioned that the pre-ticked days are determined by the workdays of the country, but is it determined by the country where the building is located, or by the region setting of the user's phone?

where the quest is located

I think it is defined at https://github.com/streetcomplete/countrymetadata/blob/master/data/workweekDays.yml

**Reply by Helium314** [reply]
> When picking the hour, clicking one of the values in the inner circle (so values from 13 to 23 and 00) almost always results in a different one getting selected.

I also have this way too often. It used to be better in previous versions (long ago)... I think it was changed either as part of upgrading dependencies, or as part so solving some issue.

**Reply by mpeter50** [reply]
> where the quest is located
> 
> I think it is defined at https://github.com/streetcomplete/countrymetadata/blob/master/data/workweekDays.yml

It's in Hungary, which is usually abbreviated as HU. I don't see it defined in that file. Maybe something else is going on here?
Also, I just checked the Region setting of the OS, and that is set as Hungary too, so this is probably not related to the OS region.

> I think it was changed either as part of upgrading dependencies, or as part so solving some issue.

Oh, so the reason I wasn't able to reproduce it is that it got fixed. That's good to hear, thanks!

**Reply by westnordost** [reply]
Yes, @matkoniecz linked the wrong file: https://github.com/streetcomplete/countrymetadata/blob/master/data/regularShoppingDays.yml
